yarn-project/end-to-end/src/bench/tx_stats_bench.test.ts

```diff
@@ -175,4 +179,88 @@ describe('transaction compression', () => {
     },
     TIMEOUT,
   );
+
+  it(
+    'verifies a single transaction',
+    async () => {
+      const numIterations = 20;
+      const resultsArray: (IVCProofVerificationResult | undefined)[] = Array.from({ length: numIterations }).map(
+        x => undefined,
+      );
+      for (let i = 0; i < numIterations; i++) {
+        const result = await t.circuitProofVerifier?.verifyProof(i % 2 === 0 ? privateProvenTx : publicProvenTx);
+        resultsArray[i] = result;
+      }
+      resultsArray.forEach(result => expect(result?.valid ?? false).toBe(true));
+      const durations = resultsArray.filter(result => result !== undefined).map(result => result.durationMs);
+      const totalDurations = resultsArray.filter(result => result !== undefined).map(result => result.totalDurationMs);
+      results.push({
+        name: `Tx IVC Verification/Single Transaction/Verification Time`,
+        value: durations.reduce((a, b) => a + b, 0) / numIterations,
+        unit: 'ms',
+      });
+      results.push({
+        name: `Tx IVC Verification/Single Transaction/Total Verification Time`,
+        value: totalDurations.reduce((a, b) => a + b, 0) / numIterations,
+        unit: 'ms',
+      });
+    },
+    TIMEOUT,
+  );
+
+  it(
+    'verifies transactions at 10 TPS',
+    async () => {
+      const seconds = 60;
+      const proofsPerSecond = 10;
+      const totalNumProofs = seconds * proofsPerSecond;
+      const promises = [];
+      const timer = new Timer();
+      for (let i = 0; i < totalNumProofs; i++) {
+        promises.push(t.circuitProofVerifier?.verifyProof(i % 2 === 0 ? privateProvenTx : publicProvenTx));
+        await sleep(1000 / proofsPerSecond);
+      }
+      const resultsArray = await Promise.all(promises);
+      const totalDuration = timer.ms();
+      resultsArray.forEach(result => expect(result?.valid ?? false).toBe(true));
+      const durations = resultsArray.filter(result => result !== undefined).map(result => result.durationMs);
+      const totalDurations = resultsArray.filter(result => result !== undefined).map(result => result.totalDurationMs);
```

Comment:
This looks fine for our benchmarks. If need to we could also switch to the [Histogram](https://nodejs.org/api/perf_hooks.html#class-recordablehistogram-extends-histogram) class in nodejs's stdlib